lib/js/stanford_person.behavior.js

```diff
@@ -21,6 +21,25 @@ export default {
           $(this).attr('aria-expanded', 'false');
         }
       });
+
+      // Looking at the h3 in the grid and hiding duplicates
+      var headingLabels = $('.stanford-people-grid--filters h3');
+      headingLabels.each(function (i) {
+        var headingLabelsCurrent = $(this).text();
+        var prev = headingLabels.eq(i - 1).text();
+
+        if (headingLabelsCurrent !== prev) {
+          $(this).show();
+        }
+        // to make sure the very first appears.
+        else if (i === 0) {
+          $(this).show();
+        }
+        else {
+          $(this).hide();
+        }
```

Comment:
All this logic can be simplified. Try this suggestion.
```suggestion
        if (headingLabelsCurrent === prev && i > 0) {
          $(this).hide();
        }
```

Reply:
Thank you! I updated it and tried on two different taxonomies.